### ISSUE TITLE
accuracy always equal to 1 when running resnet50_trainer.py

### ISSUE THREAD

**Issue by neobarney** [facebookarchive/caffe2#810]
Is it normal that the accuracy is always equal to 1, no matter the dataset (i tried MNIST, the boat/car tutorial, and custom datasets), or number of gpus ?
I tried to fetch the Blobs from the workspace and accuracy is always equal to one for all gpus
Thanks

Here is my log :

python ./../examples/resnet50_trainer.py --train_data /caffe2/caffe2/python/tutorials/tutorial_data/mnist/mnist-train-nchw-leveldb --test_data /caffe2/caffe2/python/tutorials/tutorial_data/mnist/mnist-test-nchw-leveldb --gpus 0 --batch_size 32 --num_labels 10 --epoch_size 50000 --num_epoch 10 --db_type leveldb --base_learning_rate 0.10

Ignoring @/caffe2/caffe2/contrib/nccl:nccl_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/contrib/gloo:gloo_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/contrib/gloo:gloo_ops_gpu as it is not a valid file.
Ignoring @/caffe2/caffe2/distributed:file_store_handler_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/distributed:redis_store_handler_ops as it is not a valid file.
INFO:resnet50_trainer:Running on GPUs: [0]
INFO:resnet50_trainer:Using epoch size: 49984
INFO:data_parallel_model:Parallelizing model for devices: [0]
INFO:data_parallel_model:Create input and model training operators
INFO:data_parallel_model:Model for GPU: 0
INFO:data_parallel_model:Adding gradient operators
INFO:data_parallel_model:Add gradient all-reduces for SyncSGD
INFO:data_parallel_model:Post-iteration operators for updating params
INFO:data_parallel_model:Calling optimizer builder function
INFO:data_parallel_model:Add initial parameter sync
WARNING:data_parallel_model:------- DEPRECATED API, please use data_parallel_model.OptimizeGradientMemory() ----- 
WARNING:memonger:NOTE: Executing memonger to optimize gradient memory
INFO:memonger:Remapping 128 blobs, using 19 shared
INFO:memonger:Memonger memory optimization took 0.0081160068512 secs
INFO:resnet50_trainer:----- Create test net ----
INFO:data_parallel_model:Parallelizing model for devices: [0]
INFO:data_parallel_model:Create input and model training operators
INFO:data_parallel_model:Model for GPU: 0
INFO:data_parallel_model:Parameter update function not defined --> only forward
INFO:resnet50_trainer:Starting epoch 0/10
INFO:resnet50_trainer:Finished iteration 1/1562 of epoch 0 (19.77 images/sec)
INFO:resnet50_trainer:Training loss: 2.30258488655, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 2/1562 of epoch 0 (40.52 images/sec)
INFO:resnet50_trainer:Training loss: 2.13329339027, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 3/1562 of epoch 0 (40.44 images/sec)
INFO:resnet50_trainer:Training loss: 1.90181255341, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 4/1562 of epoch 0 (40.45 images/sec)
INFO:resnet50_trainer:Training loss: 1.62707042694, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 5/1562 of epoch 0 (40.42 images/sec)
INFO:resnet50_trainer:Training loss: 1.33165848255, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 6/1562 of epoch 0 (40.39 images/sec)
INFO:resnet50_trainer:Training loss: 1.04083251953, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 7/1562 of epoch 0 (40.44 images/sec)
INFO:resnet50_trainer:Training loss: 0.778668045998, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 8/1562 of epoch 0 (40.44 images/sec)
INFO:resnet50_trainer:Training loss: 0.56205111742, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 9/1562 of epoch 0 (40.44 images/sec)
INFO:resnet50_trainer:Training loss: 0.396477043629, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 10/1562 of epoch 0 (40.48 images/sec)
INFO:resnet50_trainer:Training loss: 0.277258038521, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 11/1562 of epoch 0 (40.45 images/sec)
INFO:resnet50_trainer:Training loss: 0.194602370262, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 12/1562 of epoch 0 (40.42 images/sec)
INFO:resnet50_trainer:Training loss: 0.138311877847, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 13/1562 of epoch 0 (40.37 images/sec)
INFO:resnet50_trainer:Training loss: 0.100091867149, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 14/1562 of epoch 0 (40.37 images/sec)
INFO:resnet50_trainer:Training loss: 0.0739600434899, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 15/1562 of epoch 0 (40.31 images/sec)
INFO:resnet50_trainer:Training loss: 0.0558631159365, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 16/1562 of epoch 0 (40.34 images/sec)
INFO:resnet50_trainer:Training loss: 0.0431294031441, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 17/1562 of epoch 0 (40.31 images/sec)
INFO:resnet50_trainer:Training loss: 0.0340112037957, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 18/1562 of epoch 0 (40.29 images/sec)
INFO:resnet50_trainer:Training loss: 0.0273668635637, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 19/1562 of epoch 0 (40.23 images/sec)
INFO:resnet50_trainer:Training loss: 0.022438922897, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 20/1562 of epoch 0 (40.35 images/sec)
INFO:resnet50_trainer:Training loss: 0.0187235400081, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 21/1562 of epoch 0 (40.30 images/sec)
INFO:resnet50_trainer:Training loss: 0.0158774871379, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 22/1562 of epoch 0 (40.30 images/sec)
INFO:resnet50_trainer:Training loss: 0.0136659229174, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 23/1562 of epoch 0 (40.18 images/sec)
INFO:resnet50_trainer:Training loss: 0.0119224116206, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 24/1562 of epoch 0 (40.11 images/sec)
INFO:resnet50_trainer:Training loss: 0.0105315642431, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 25/1562 of epoch 0 (40.31 images/sec)
INFO:resnet50_trainer:Training loss: 0.00940877292305, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 26/1562 of epoch 0 (40.21 images/sec)
INFO:resnet50_trainer:Training loss: 0.00849325582385, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 27/1562 of epoch 0 (40.23 images/sec)
INFO:resnet50_trainer:Training loss: 0.0077387271449, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 28/1562 of epoch 0 (40.30 images/sec)
INFO:resnet50_trainer:Training loss: 0.0071104131639, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 29/1562 of epoch 0 (40.15 images/sec)
INFO:resnet50_trainer:Training loss: 0.00658509042114, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 30/1562 of epoch 0 (40.23 images/sec)
INFO:resnet50_trainer:Training loss: 0.00614053150639, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 31/1562 of epoch 0 (40.20 images/sec)
INFO:resnet50_trainer:Training loss: 0.00576298404485, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 32/1562 of epoch 0 (40.16 images/sec)
INFO:resnet50_trainer:Training loss: 0.0054386514239, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 33/1562 of epoch 0 (40.25 images/sec)
INFO:resnet50_trainer:Training loss: 0.00516010867432, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 34/1562 of epoch 0 (40.26 images/sec)
INFO:resnet50_trainer:Training loss: 0.00491885701194, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 35/1562 of epoch 0 (40.27 images/sec)
INFO:resnet50_trainer:Training loss: 0.00470958091319, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 36/1562 of epoch 0 (40.35 images/sec)
INFO:resnet50_trainer:Training loss: 0.00452589476481, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 37/1562 of epoch 0 (40.27 images/sec)
INFO:resnet50_trainer:Training loss: 0.00436460925266, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 38/1562 of epoch 0 (40.38 images/sec)
INFO:resnet50_trainer:Training loss: 0.00422252668068, accuracy: 1.0
INFO:resnet50_trainer:Finished iteration 39/1562 of epoch 0 (40.19 images/sec)
INFO:resnet50_trainer:Training loss: 0.00409644981846, accuracy: 1.0

**Comment by LimingYin** [comment]
in line 50, `data, label = brew.image_input( ... `
I print the data out, it is all zeros.
I think the problem is in first step image_input.

**Comment by neobarney** [comment]
same for me, data is always all zeros

**Comment by chenriwei** [comment]
@LimingYin I checkout the label and data value with code

```
label = workspace.FetchBlob(prefix + '/label')
data = workspace.FetchBlob(prefix + '/data')
```

it result that all the data value is 0. and all the label  value is 0

**Comment by neobarney** [comment]
any updates on what's wrong with the data input in the trainer?

**Comment by neobarney** [comment]
I finally found the solution, it's simply because "use_caffe_datum" is set to True by default in the trainer, while we're trying to use caffe2 format in input.

Simply change it to "False" if you're using input data in caffe2 format !

**Comment by PhilBotSeesU** [comment]
I have the same issue with the zero's being thrown for image data but the 'use_caffe_datum=False' does not work. When I do that it throws the following. I'm using an LMDB database. I notice you say "using data in caffe2 format". 

I've confirmed that I'm building the lmdb database and each image is not a bunch of zeros. 

Any help is appreciated it. 

: python ~/resnet_te/resnet50_trainer.py --train_data=/home/ubuntu/resnet_te/data/database/train_db/ --num_gpus=4 --num_epochs=5 --epoch_size=500 --num_labels=2

...

INFO:resnet50_trainer:Starting epoch 0/5
F0708 17:08:22.785287   387 image_input_op.h:354] Unknown image data type.
*** Check failure stack trace: ***
F0708 17:08:22.786284   388 image_input_op.h:354] Unknown image data type.
*** Check failure stack trace: ***
Aborted (core dumped)
ubuntu@lambda-devbox:~/r

**Comment by xkszltl** [comment]
The image_input_op.h indicates that you can only use BYTE (for raw bitmap) or STRING (for encoded img) as input.

```
if (image_proto.data_type() == TensorProto::STRING) {...}
else if (image_proto.data_type() == TensorProto::BYTE) {...}
else LOG(FATAL) << "Unknown image data type.";
```

**Comment by gemfield** [comment]
When I print the value of workspace.FetchBlob(prefix + '/data'), I got all -1, which is different from 0 above. Quoted from https://zhuanlan.zhihu.com/p/45067948, that ImageInputOp class source code show us the use_caffe_datum usage:
```
if (use_caffe_datum_) {
    // The input is a caffe datum format.
    caffe::Datum datum;
    CAFFE_ENFORCE(datum.ParseFromString(value));
    prefetched_label_.mutable_data<int>()[item_id] = datum.label();
    ......
  } else {
    // The input is a caffe2 format.
    TensorProtos protos;
    CAFFE_ENFORCE(protos.ParseFromString(value));
    const TensorProto& image_proto = protos.protos(0);
    const TensorProto& label_proto = protos.protos(1);
    vector<TensorProto> additional_output_protos;
    ......
  }
```
